sycl/source/detail/graph_impl.cpp

```diff
@@ -820,6 +822,10 @@ exec_graph_impl::enqueue(const std::shared_ptr<sycl::detail::queue_impl> &Queue,
             Event->wait(Event);
           } else {
             CGData.MEvents.push_back(Event);
+            // Add events of the previous execution of all graph partitions.
+            for (auto &AttachedEvent : Event->getPostCompleteEvents()) {
+              CGData.MEvents.push_back(AttachedEvent);
```

Comment:
Could we reserve the space in `CGData.MEvents` before we add these? You could even do it before line 824.

Reply:
The code has been updated this way.